main_test.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"testing"
+	"time"
+)
+
+func TestRelativeTime(t *testing.T) {
+	tests := []struct {
+		item Item
+		want string
+	}{
+		{
+			item: Item{
+				// title:  "Random title",
+				// author: "Mr Crabs",
+				// score:  0,
+				time: time.Date(2024, 8, 13, 14, 30, 45, 100, time.Local),
+				// URL:    "example.com/404",
+			},
+			want: "6m",
+		},
+		{
+			item: Item{
+				time: time.Date(2025, 1, 01, 14, 30, 45, 100, time.Local),
+			},
+			want: "1m",
+		},
+		{
+			item: Item{
+				time: time.Date(2025, 2, 01, 14, 30, 45, 100, time.Local),
+			},
+			want: "12d",
+		},
+		{
+			item: Item{
+				time: time.Date(2024, 2, 01, 14, 30, 45, 100, time.Local),
+			},
+			want: "1y",
+		},
+	}
+
+	for _, test := range tests {
+		got := test.item.relativeTime()
```

Comment:
Here's an interesting problem: how can you make sure those tests will work 1 day, 1 week, 10 years from now?

The only real solution is to not have `time.Since` directly, since you can't control what the value of this function is. You can go one of 2 routes:
- in the `Item` struct, you store "hoursSincePublishing", and parse off this. And when you initialize the item (after parsing the API), you do the conversion.
- you pass to the `relativeTime()` function the value of `time.Now()`, and do the time math off this value.